ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java

```diff
@@ -1081,27 +1100,28 @@ public void subsetOfInvalidPendingTransactionsIncludedWhenTxSelectionMaxTimeIsOv
         processingTooLate,
         postProcessingTooLate,
         500,
-        BLOCK_SELECTION_TIMEOUT,
-        false,
+        BLOCK_SELECTION_TIMEOUT_INVALID_TX,
+        true,
         NONCE_TOO_LOW);
   }
 
   @ParameterizedTest
   @MethodSource("subsetOfPendingTransactionsIncludedWhenTxSelectionMaxTimeIsOver")
-  public void invalidPendingTransactionsThatTakesTooLongToEvaluateIsDroppedFromThePool(
-      final boolean isPoa,
-      final boolean preProcessingTooLate,
-      final boolean processingTooLate,
-      final boolean postProcessingTooLate) {
+  public void
+      evaluationOfInvalidPendingTransactionThatTakesTooLongToEvaluateIsInterruptedAndPenalized(
+          final boolean isPoa,
+          final boolean preProcessingTooLate,
+          final boolean processingTooLate,
+          final boolean postProcessingTooLate) {
 
     internalBlockSelectionTimeoutSimulationInvalidTxs(
         isPoa,
         preProcessingTooLate,
         processingTooLate,
         postProcessingTooLate,
         900,
-        INVALID_TX_EVALUATION_TOO_LONG,
```

Comment:
I think `INVALID_TX_EVALUATION_TOO_LONG` becomes missing from the tests. In which scenario we can receive this error?

Reply:
this is a very edge case, that could happen when the tx is invalid and complete the processing while at the same time the interrupt occurs, and it is not possible to cover without first making this test deterministic.

Reply:
From my examination of the code with the same config used to limit block building and transaction execution, I believe it can't happen at all. But it might make sense to separate these two in the future